test/customer/postCustomerLoginIT.js

```diff
@@ -147,6 +148,90 @@ describe('magento postCustomerLogin', function() {
                 });
         });
 
+        it('successfully login a customer & merge a cart', function() {
+            let anonymousCartId;
+            let accessToken;
+            let customerCartId;
+            //creates an anonymous cart
+            return chai.request(env.openwhiskEndpoint)
+                .post(env.cartsPackage + 'postCartEntry')
+                .query({
+                    currency: 'EUR',
+                    quantity: 1,
+                    productVariantId: productVariantEqbisumas11
+
+                })
+                //verify anonymous cart
+                .then(response => {
+                    expect(response.body.id).to.not.be.undefined;
+                    expect(response.body.entries).to.have.lengthOf(1);
+                    anonymousCartId = response.body.id;
+                })
+                //post a customer login without merge
+                .then( () => {
```

Comment:
This is probably not doing what you want. You can register only one callback for a single promise. This means the `then` block in L164-168 is never executed.

Reply:
hmm, yes the correct should be to return a promise, but it's working. did you test it on your env?

Reply:
Well, it might work. Check the minimal example `Promise.resolve("a").then((res) => console.log("1", res)).then((res) => console.log("2", res));`, both `then` statements are executed. I would still prefer to have only one `then` per Promise, it makes the code much cleaner.

Reply:
yes, I know what it does. As long as the second res is not used there's no problem. I agree we can return a promise just to make things clear.

Reply:
What about you just remove the unnecessary `then`?